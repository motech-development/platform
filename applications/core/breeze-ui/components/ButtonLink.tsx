import { FC, forwardRef, HTMLProps } from 'react';

const ButtonLink: FC<HTMLProps<HTMLButtonElement>> =
  forwardRef<HTMLButtonElement>((props, ref) => (
    <button
      className="appearance-none bg-none border-0 font-semibold text-blue-600 hover:text-red-600 hover:underline focus:outline-none"
      {...props}
      ref={ref}
    />
  ));

export default ButtonLink;
