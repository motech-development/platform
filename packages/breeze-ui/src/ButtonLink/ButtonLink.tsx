import { FC, ButtonHTMLAttributes, memo } from 'react';
import BaseLink from '../BaseLink/BaseLink';

const ButtonLink: FC<ButtonHTMLAttributes<HTMLButtonElement>> = (props) => (
  // eslint-disable-next-line react/jsx-props-no-spreading
  <BaseLink as="button" {...props} />
);

export default memo(ButtonLink);
