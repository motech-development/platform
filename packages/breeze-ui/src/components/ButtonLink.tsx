import { ButtonHTMLAttributes, FC, DetailedHTMLProps } from 'react';
import Link from './Link';

export type TButtonLink = DetailedHTMLProps<
  ButtonHTMLAttributes<HTMLButtonElement>,
  HTMLButtonElement
>;

// eslint-disable-next-line react/jsx-props-no-spreading
const ButtonLink: FC<TButtonLink> = (props) => <Link as="button" {...props} />;

export default ButtonLink;
