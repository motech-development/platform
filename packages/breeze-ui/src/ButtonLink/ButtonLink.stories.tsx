import { withA11y } from '@storybook/addon-a11y';
import { text, withKnobs } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react';
import React from 'react';
import BaseStyles from '../BaseStyles/BaseStyles';
import Card from '../Card/Card';
import ButtonLink from './ButtonLink';

const stories = storiesOf('ButtonLink', module);

stories.addDecorator(withA11y);
stories.addDecorator(withKnobs);

stories.add('Basic button link', () => (
  <>
    <BaseStyles />

    <Card>
      <ButtonLink>{text('Text', 'This is a button')}</ButtonLink>
    </Card>
  </>
));
