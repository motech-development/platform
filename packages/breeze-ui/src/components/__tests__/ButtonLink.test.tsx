import { render } from '@testing-library/react';
import ButtonLink from '../ButtonLink';

describe('ButtonLink', () => {
  it('should correctly render the component', () => {
    const { container } = render(<ButtonLink type="submit">Button</ButtonLink>);

    expect(container.firstChild).toMatchSnapshot();
  });
});
