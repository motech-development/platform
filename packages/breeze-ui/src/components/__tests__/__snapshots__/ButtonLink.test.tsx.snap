// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonLink should correctly render the component 1`] = `
<button
  class="appearance-none bg-none border-0 font-semibold text-blue-600 underline hover:no-underline hover:text-red-600 focus:outline-none"
  type="submit"
>
  Button
</button>
`;
